components/server/src/bitbucket-server/bitbucket-server-api.ts

```diff
@@ -391,20 +404,20 @@ export class BitbucketServerApi {
         };
 
         if (query.searchString?.trim()) {
-            const result: BitbucketServer.Repository[] = [];
-            const ids = new Set<number>(); // used to deduplicate
-            for (const param of ["name", "projectname"]) {
-                const pageResult = await runQuery(`${param}=${query.searchString}`);
-                for (const repo of pageResult) {
-                    if (!ids.has(repo.id)) {
-                        ids.add(repo.id);
-                        result.push(repo);
-                    }
-                }
+            const results: Map<number, BitbucketServer.Repository> = new Map();
+
+            // Query by name & projectname in parrallel
+            const [nameResults, projectResults] = await Promise.all([
```

Comment:
I refactored this so we issue both calls in parallel

Reply:
Was there a perf issue? Asking because downside can be burning of API request limits faster. I understand it is fine with incremental, but without feature flag it won't be a case.

Reply:
I didn't notice an issue with perf so I can swap it back, this just seemed like a better approach, but you're right, it would make api calls quicker. We should only be fetching 1 page of each request in this scenario though, but that's only really enforced by someone calling this method correctly, i.e. providing a search string, and setting limit/cap to something low.

Part of it might be this method could be adjusted so callers didn't have as much say in how we fetch repos (and how many pages to get), but instead just provided a limit, and we fetched up until we got that limit, which could be a single page in many cases. I didn't want to make that big of a change here atm though as I'm not super familiar with it yet.

Reply:
Changed it to be in series.